tests/sentry/workflow_engine/migrations/test_0034_migrate_issue_alerts.py

```diff
@@ -0,0 +1,379 @@
+from sentry.constants import ObjectStatus
+from sentry.grouping.grouptype import ErrorGroupType
+from sentry.models.rulesnooze import RuleSnooze
+from sentry.rules.age import AgeComparisonType
+from sentry.rules.conditions.event_frequency import EventUniqueUserFrequencyConditionWithConditions
+from sentry.rules.conditions.every_event import EveryEventCondition
+from sentry.rules.conditions.reappeared_event import ReappearedEventCondition
+from sentry.rules.conditions.regression_event import RegressionEventCondition
+from sentry.rules.filters.age_comparison import AgeComparisonFilter
+from sentry.testutils.cases import TestMigrations
+from sentry.testutils.helpers import install_slack
+from sentry.workflow_engine.migration_helpers.issue_alert_migration import IssueAlertMigrator
+from sentry.workflow_engine.models import (
+    Action,
+    AlertRuleDetector,
+    AlertRuleWorkflow,
+    DataCondition,
+    DataConditionGroup,
+    DataConditionGroupAction,
+    Detector,
+    DetectorWorkflow,
+    Workflow,
+    WorkflowDataConditionGroup,
+)
+from sentry.workflow_engine.models.data_condition import Condition
+
+
+class TestMigrateIssueAlerts(TestMigrations):
+    migrate_from = "0033_workflow_name_256_char"
+    migrate_to = "0034_migrate_issue_alerts"
+    app = "workflow_engine"
+
+    def setup_initial_state(self):
+        conditions = [
+            {"id": ReappearedEventCondition.id},
+            {"id": RegressionEventCondition.id},
+            {
+                "id": AgeComparisonFilter.id,
+                "comparison_type": AgeComparisonType.OLDER,
+                "value": "10",
+                "time": "hour",
+            },
+        ]
+        integration = install_slack(self.organization)
+        self.action_data = [
+            {
+                "channel": "#my-channel",
+                "id": "sentry.integrations.slack.notify_action.SlackNotifyServiceAction",
+                "workspace": str(integration.id),
+                "uuid": "test-uuid",
+                "channel_id": "C01234567890",
+            },
+        ]
+        self.issue_alert = self.create_project_rule(
+            name="test1",
+            condition_data=conditions,
+            action_match="any",
+            filter_match="any",
+            action_data=self.action_data,
+            frequency=5,
+        )
+
+        self.issue_alert_missing_matches = self.create_project_rule(
+            name="test2",
+            condition_data=conditions,
+            frequency=5,
+            action_data=self.action_data,  # TODO: comment out to test sentry apps
```

Comment:
can't merge until we finalize how we will avoid an RPC call for sentry apps inside a transaction in the migration